### PR TITLE
Update developer guide

### PR DESCRIPTION
**Pull request by cwperks**
### Description

Updates the developer guide which had many outdated references to 1.x.

### Category
[Enhancement, New feature, Bug fix, Test fix, Refactoring, Maintenance, Documentation]

Documentation

### Check List
- [ ] New functionality includes testing
- [ ] New functionality has been documented
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: DEVELOPER_GUIDE.md [hunk 1]
```diff
@@ -7,74 +8,51 @@ So you want to contribute code to this project? Excellent! We're glad you're her
...
+As a prerequisite, please follow [the developer guide of the Security Plugin](https://github.com/opensearch-project/security/blob/main/DEVELOPER_GUIDE.md). This will get an OpenSearch node running with security plugin installed and using a demo configuration. 
 
-At present there are following branches available to choose from for the setup:
+At present, there are the following branches available to choose from for the setup:
 
 ### **Back-end**
 
 | OpenSearch<br>branch | Security Plugin<br>branch  | OpenSearch<br>version  |
 |--------              |---                         |---                  |
-| [1.x](https://github.com/opensearch-project/OpenSearch/tree/1.x) | [main](https://github.com/opensearch-project/security/) | [v1.3.0](https://github.com/opensearch-project/OpenSearch/blob/6eda740be744846f7aa0b2674820b5ed9b6be17e/buildSrc/version.properties#L1) |
```
Comment (scrawfor99) [inline DEVELOPER_GUIDE.md:24]: Did you mean to remove 1.x?

File: DEVELOPER_GUIDE.md [hunk 2]
```diff
@@ -7,74 +8,51 @@ So you want to contribute code to this project? Excellent! We're glad you're her
...
 
 ### **Front-end**
 | OpenSearch Dashboards<br>branch | Security Dashboards Plugin<br>branch  | OpenSearch Dashboards<br>version  |
 | ---         | ---                         | ---                  |
-| [1.x](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/1.x) | [main](https://github.com/opensearch-project/security-dashboards-plugin/) | [v1.3.0](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/b82f214babf7ef8db4a9705b7b51a912f779184c/package.json#L14) |
-| [main](https://github.com/opensearch-project/OpenSearch-Dashboards) | (under development) | [v2.0.0](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/0c3f901c569838b2708a079d058c7d09970b944c/package.json#L14) |
+| [1.x](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/1.3) | [1.3](https://github.com/opensearch-project/security-dashboards-plugin/tree/1.3) | [v1.3.x](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/1.x/package.json#L14) |
```
Comment (scrawfor99) [inline DEVELOPER_GUIDE.md:28]: See above, you have 1.x here but not for backend.

**Review by scrawfor99 — COMMENTED** [review]
Looks good, just a quick version question. Thank you for this.

File: DEVELOPER_GUIDE.md (hunk 1 above)
Comment (cwperks) [reply DEVELOPER_GUIDE.md:24]: Yes, I replaced it with 1.3 since that is the LTS branch on the 1.X line.

File: DEVELOPER_GUIDE.md (hunk 2 above)
Comment (cwperks) [reply DEVELOPER_GUIDE.md]: Changed to 1.3.

**Review by scrawfor99 — APPROVED** [review]
Great job! Thank you for fixing this up.

**Review by DarshitChanpura — APPROVED** [review]

**Review by DarshitChanpura — COMMENTED** [review]
ty @cwperks for updating this!

**Pull request merged**